File: LeetCodeProblemsJava/src/Problems/LeetCode2574LeftAndRightSumDifferences.java
package Problems;

public class LeetCode2574LeftAndRightSumDifferences {
    public int[] leftRightDifference(int[] nums) {
        int length = nums.length;
        int[] left = new int[length];
        int[] right = new int[length];
        for (int i = 1; i < length; i++) {
            left[i]=left[i-1]+nums[i-1];
            right[length-i-1]=right[length-i]+nums[length-i];
        }
        for (int i = 0; i < length; i++) {
            nums[i] = Math.abs(left[i]-right[i]) ;
        }
        return nums;
     }
}

File: LeetCodeProblemsGo/LeetCode2574LeftAndRightSumDifferences/LeetCode2574LeftAndRightSumDifferences.go
package LeetCode2574LeftAndRightSumDifferences

import "math"

func leftRightDifference(nums []int) []int {
	length := len(nums)
	left := make([]int, length)
	right := make([]int, length)
	for i := 1; i < length; i++ {
		left[i] = left[i-1] + nums[i-1]
		right[length-i-1] = right[length-i] + nums[length-i]
	}
	for i := 0; i < length; i++ {
		nums[i] = int(math.Abs(float64(left[i] - right[i])))
	}
	return nums
}
